types/list_test.go

```diff
@@ -361,16 +362,32 @@ func TestListToTerraformValue(t *testing.T) {
 				tftypes.NewValue(tftypes.String, "hello, world"),
 			}),
 		},
+		"no-elem-type": {
+			input: List{
+				Elems: []attr.Value{
+					String{Null: true},
+					String{Value: "hello, world"},
+				},
+			},
+			expectedErr: "cannot convert List to Terraform Value if ElemType field is null",
+			expectation: tftypes.Value{},
+		},
 	}
 	for name, test := range tests {
 		name, test := name, test
 		t.Run(name, func(t *testing.T) {
 			t.Parallel()
 
-			got, err := test.input.ToTerraformValue(context.Background())
-			if err != nil {
-				t.Errorf("Unexpected error: %s", err)
-				return
+			got, gotErr := test.input.ToTerraformValue(context.Background())
+			if gotErr != nil {
```

Comment:
Nit: It'd be good to check against `gotErr == nil && test.expectedErr != nil` after this `if` conditional as well.

Another, potentially simpler way to test this is via `cmp.Diff()` as well.

Reply:
I have updated all the tests I added to check the result this way:

```golang
			if test.expectedErr == "" && gotErr != nil {
				t.Errorf("Unexpected error: %s", gotErr)
				return
			}

			if test.expectedErr != "" {
				if gotErr == nil {
					t.Errorf("Expected error to be %q, got none", test.expectedErr)
					return
				}

				if test.expectedErr != gotErr.Error() {
					t.Errorf("Expected error to be %q, got %q", test.expectedErr, gotErr.Error())
					return
				}
			}
```

This should cover all combinations we want to spot.